### PR TITLE
MaximalSubgroupClassReps uses AtlasRep if available

### PR DESCRIPTION
**Pull request by hulpke**
MaximalSubgroupClassReps method for simple groups that uses AtlasRep, if thee package is available and can handle the group.

### CODE REVIEW

**Review by fingolfin — APPROVED** [review]
Thank you! Looks good to me but it wouldn't hurt if @ThomasBreuer had another look, perhaps?

**Review by ThomasBreuer — CHANGES_REQUESTED** [review]
The idea is:
If AtlasRep provides information about the group in question then does it have SLPs for computing representatives for all classes of maximal subgroups?
In order to decide this, one can do the following.
```
info:= AtlasRepInfoRecord( name );
IsBound( info.nrMaxes ) and info.slpMaxes[1] = [ 1 .. info.nrMaxes ] and ForAll( info.slpMaxes[2], l -> 1 in l );
```
If the result is `true` then one can proceed as in the proposed code: Compute standard generators and use the SLPs directly if there is a `find` script, and compute an isomorphisms of groups and use `AtlasSubgroup` otherwise.

File: lib/grplatt.gi [hunk 1]
```diff
@@ -3132,6 +3132,35 @@ local recog,m;
...
+    if IsSimpleGroup(G) and IsPackageMarkedForLoading("atlasrep","")=true
+      and ValueOption(NO_PRECOMPUTED_DATA_OPTION)<>true then
+      recog:=DataAboutSimpleGroup(G);
+      inf:=CallFuncList(ValueGlobal("AtlasRepInfoRecord"),[recog.tomName]);
+      p:=CallFuncList(ValueGlobal("AtlasProgram"),[recog.tomName,1,"find"]);
+      if p<>fail and inf<>fail and IsBound(inf.nrMaxes) then
+        Info(InfoLattice,1,"Maxes of ",recog.tomName," by ATLAS words");
+        a:=CallFuncList(ValueGlobal("ResultOfBBoxProgram"),[p.program,G]);
+        p:=List([1..inf.nrMaxes],
+          x->CallFuncList(ValueGlobal("AtlasProgram"),[inf.name,1,"maxes",x]));
```
[the remark this answers is not in the document]
Comment (hulpke) [reply lib/grplatt.gi:3146]: Thank you -added

**Comment by fingolfin** [comment]
@ThomasBreuer does this seem OK to you now?

File: lib/grplatt.gi [hunk 2]
```diff
@@ -3132,6 +3132,41 @@ local recog,m;
...
+    if IsSimpleGroup(G) and IsPackageMarkedForLoading("atlasrep","")=true
+      and ValueOption(NO_PRECOMPUTED_DATA_OPTION)<>true then
+      recog:=DataAboutSimpleGroup(G);
+      inf:=CallFuncList(ValueGlobal("AtlasRepInfoRecord"),[recog.tomName]);
+      if inf<>fail and IsBound( inf.nrMaxes )
+          and IsBound(inf.slpMaxes)
+          and inf.slpMaxes[1] = [ 1 ..  inf.nrMaxes ]
+          and ForAll( inf.slpMaxes[2], l -> 1 in l ) then
+        p:=CallFuncList(ValueGlobal("AtlasProgram"),[recog.tomName,1,"find"]);
+        if p<>fail and inf<>fail and IsBound(inf.nrMaxes) then
```
Comment (ThomasBreuer) [inline lib/grplatt.gi:3146]:
```suggestion
        if p<>fail then
```

File: lib/grplatt.gi [hunk 3]
```diff
@@ -3132,6 +3132,41 @@ local recog,m;
...
+          Info(InfoLattice,1,"Maxes of ",recog.tomName," by ATLAS words");
+          a:=CallFuncList(ValueGlobal("ResultOfBBoxProgram"),[p.program,G]);
+          p:=List([1..inf.nrMaxes],
+            x->CallFuncList(ValueGlobal("AtlasProgram"),[inf.name,1,"maxes",x]));
+          if ForAny(p,x->x=fail) then return fail;fi;
+          m:=List(p,x->CallFuncList(ValueGlobal("ResultOfStraightLineProgram"),
+            [x.program,a]));
+          m:=List(m,x->SubgroupNC(G,x));
+          return m;
+        elif inf<>fail and IsBound(inf.nrMaxes) then
```
Comment (ThomasBreuer) [inline lib/grplatt.gi:3156]:
```suggestion
        else
```

**Review by ThomasBreuer — APPROVED** [review]
Thanks.
The code looks correct now, just two lines could be simplified.

**Comment by fingolfin** [comment]
Thank you! I applied @ThomasBreuer tweaks (after double checking), this will auto-merge once the test suite passed

**Pull request merged**